### DISCUSSION TITLE
Simulating depletion interaction explicitly using NμpVT ensemble

### DISCUSSION THREAD

**Discussion by sauravIISc** [glotzerlab/hoomd-blue#1674] [Support] [answered]
Hi all,

I am new to HOOMD and wanted to simulate spherocylinders mediated by depletion interaction through penetrable spheres. As implicit depletion interaction gives incorrect results in the current version of hoomd I am trying to model it explicitly. I wanted to know after defining two types of particles in my system under the same shape integrator(`hoomd.hpmc.integrate.ConvexSpheropolyhedron() `) can one of them be randomly inserted and deleted, and the number of other particle type remain fixed. 

I mean let's say 'A' is the colloid whose number remains fixed in my system and let 'B' be the polymer responsible for depletion interaction whose number will fluctuate about an average number. Both the particles are in a box of fixed volume. Can I define grand canonical ensemble for only 'B'? 

Also to make potential between 'B-B' zero and infinity when 'A-A' or 'A-B' overlaps, I was thinking of defining an interaction matrix for them.  

Any help to clarify the workflow and algorithm is appreciated. Thanking all.

Regards,
Saurav.

**Comment by joaander** [comment, marked as answer]
Use the `transfer_types` argument of [`MuVT`](https://hoomd-blue.readthedocs.io/en/v4.4.0/module-hpmc-update.html#hoomd.hpmc.update.MuVT) to select which particles to insert and delete.
```
transfer_types = ['B']
```

Set elements of [`HPMCIntegrator.interaction_matrix`](https://hoomd-blue.readthedocs.io/en/v4.4.0/module-hpmc-integrate.html#hoomd.hpmc.integrate.HPMCIntegrator.interaction_matrix) to implement the penetrable hard sphere model.
```
integrator.interaction_matrix[('B', 'B')] = False
```

**Reply by sauravIISc** [reply]
Thank you so much for your prompt reply and clarification.

Regards,
Saurav.
